examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,244 @@
+import type { DataSource, syncCollection } from "./data"
+
+import { ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { useState, useMemo, useLayoutEffect } from "react"
+import { computeFieldsFromDataSource, mergeFieldsWithExistingFields } from "./data"
+import { Spinner } from "./components/Spinner"
+import { UI_DEFAULTS } from "./constants"
+
+interface FieldMappingRowProps {
+    originalField: ManagedCollectionField
+    field: ManagedCollectionField
+    isIgnored: boolean
+    onFieldToggle: (fieldId: string) => void
+    onFieldNameChange: (fieldId: string, name: string) => void
+}
+
+function FieldMappingRow({ originalField, field, isIgnored, onFieldToggle, onFieldNameChange }: FieldMappingRowProps) {
+    const isUnsupported = !field
+    const hasFieldNameChanged = field!.name !== originalField.name
+    const fieldName = hasFieldNameChanged ? field!.name : ""
+    const placeholder = isUnsupported ? "Unsupported Field" : originalField.name
+    const isDisabled = isUnsupported || isIgnored
+
+    return (
+        <>
+            <div
+                className="source-field"
+                aria-disabled={isDisabled}
+                onClick={() => onFieldToggle(field!.id)}
+                role="button"
+            >
+                <input
+                    type="checkbox"
+                    disabled={isUnsupported}
+                    checked={!isDisabled}
+                    onChange={event => {
+                        event.stopPropagation()
+                    }}
+                />
+                <span>{originalField.name}</span>
+            </div>
+            <svg xmlns="http://www.w3.org/2000/svg" width="8" height="16">
+                <path
+                    d="M 3 11 L 6 8 L 3 5"
+                    fill="transparent"
+                    strokeWidth="1.5"
+                    stroke="#999"
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                ></path>
+            </svg>
+            <input
+                type="text"
+                style={{
+                    width: "100%",
+                    opacity: isDisabled ? 0.5 : 1,
+                }}
+                disabled={isDisabled}
+                placeholder={placeholder}
+                value={fieldName || ""}
+                onChange={event => {
+                    if (!field) return
+
+                    const value = event.target.value
+                    if (!value.trim()) {
+                        onFieldNameChange(field.id, originalField.name)
+                    } else {
+                        onFieldNameChange(field.id, value.trimStart())
+                    }
```

Comment:
Let's trim on line 64, so you don't need to do that on the other lines

Reply:
We need both `trim()` and `trimStart()` here:
  - trim() in the condition checks if the input is all whitespace
  - trimStart() in the value ensures no leading spaces

We can't just use `trim()` everywhere as it would prevent trailing spaces, which I think we want to allow (e.g., "Test A")

Alternatively, I can trim when calling `collection.setFields`?